Bottom: 068f8685f94d5418e3eda7fddc0f92c0e4742055
Top:    9283f42bc436a02871e8161e68cb1895f7a77e4f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-10 14:23:02 +0000

Refresh of add-raw-polygons-support-for-t

---

diff --git a/src/hid/common/hidgl.c b/src/hid/common/hidgl.c
index 7106410..374fe5e 100644
--- a/src/hid/common/hidgl.c
+++ b/src/hid/common/hidgl.c
@@ -645,12 +645,13 @@ hidgl_fill_pcb_polygon (PolygonType *poly)
     return;
   }
 
+  /* JUST DRAW THE FIRST PIECE */
   /* Walk the polygon structure, counting vertices */
   piece = poly->Clipped;
-  do {
+//  do {
     for (contour = piece->contours; contour != NULL; contour = contour->next)
       vertex_count += contour->Count;
-  } while ((piece = piece->f) != poly->Clipped);
+//  } while ((piece = piece->f) != poly->Clipped);
 
   vertices = malloc (sizeof(GLdouble) * vertex_count * 3);
 
@@ -662,11 +663,12 @@ hidgl_fill_pcb_polygon (PolygonType *poly)
 
   gluTessBeginPolygon (tobj, NULL);
 
+  /* JUST DRAW THE FIRST PIECE */
   /* Walk the polygon structure, adding the vertices */
   i = 0;
   cc = 1;
   piece = poly->Clipped;
-  do {
+//  do {
     for (contour = piece->contours; contour != NULL; contour = contour->next) {
       gluTessBeginContour (tobj);
 
@@ -683,7 +685,7 @@ hidgl_fill_pcb_polygon (PolygonType *poly)
       } while ((vnode = vnode->next) != &contour->head);
       gluTessEndContour (tobj);
     }
-  } while ((piece = piece->f) != poly->Clipped);
+//  } while ((piece = piece->f) != poly->Clipped);
 
   gluTessEndPolygon (tobj);
   gluDeleteTess (tobj);
